cmd/bmp180/main.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2017 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// bmp180 reads the current temperature and pressure from a BMP180.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/conn/pin"
+	"periph.io/x/periph/conn/pin/pinreg"
+	"periph.io/x/periph/devices"
+	"periph.io/x/periph/experimental/devices/bmp180"
+	"periph.io/x/periph/host"
+)
+
+func printPin(fn string, p pin.Pin) {
+	name, pos := pinreg.Position(p)
+	if name != "" {
+		log.Printf("  %-4s: %-10s found on header %s, #%d\n", fn, p, name, pos)
+	} else {
+		log.Printf("  %-4s: %-10s\n", fn, p)
+	}
+}
+
+func read(e devices.Environmental, interval time.Duration) error {
+	var t *time.Ticker
+	if interval != 0 {
+		t = time.NewTicker(interval)
+	}
+
+	for {
+		var env devices.Environment
+		if err := e.Sense(&env); err != nil {
+			return err
+		}
+		fmt.Printf("%8s %10s %9s\n", env.Temperature, env.Pressure, env.Humidity)
+		if t == nil {
+			break
+		}
+
+		<-t.C
+	}
+	return nil
+}
+
+func mainImpl() error {
+	i2cID := flag.String("i2c", "", "I²C bus to use")
+	hz := flag.Int("hz", 0, "I²C bus speed")
```

Comment:
I'd say to not bother, I had added these because I was experimenting but it's really irrelevant for a sensing device, I'll post a change to remove it from the bme280 exec.

Also setspeed is not working fine on the rpi so it's not even usable at the time. :(